dalek/cleaners/resource_group_palo_alto_local_rulestack.go

```diff
@@ -0,0 +1,117 @@
+	"github.com/tombuildsstuff/azurerm-dalek/clients"
+)
+
+type PaloAltoLocalRulestackCleaner struct{}
+
+var _ ResourceGroupCleaner = PaloAltoLocalRulestackCleaner{}
+
+func (p PaloAltoLocalRulestackCleaner) Name() string {
+	return "Removing Rulestack Rules"
+}
+
+func (p PaloAltoLocalRulestackCleaner) Cleanup(ctx context.Context, id commonids.ResourceGroupId, client *clients.AzureClient) error {
+	rulestacksClient := client.ResourceManager.PaloAltoLocalRulestacksClient
+
+	rulestacks, err := rulestacksClient.ListByResourceGroupComplete(ctx, id)
+	if err != nil {
+		return err
+	}
+
+	// Rules
+	rulesClient := client.ResourceManager.PaloAltoLocalRulestackRuleClient
+	for _, rg := range rulestacks.Items {
+		rulestackId := localrules.NewLocalRulestackID(id.SubscriptionId, id.ResourceGroupName, pointer.From(rg.Name))
+		rulesInRulestack, err := rulesClient.ListByLocalRulestacksComplete(ctx, rulestackId)
+		if err != nil {
+			return fmt.Errorf("listing rules for %s: %+v", id, err)
+		}
+		for _, v := range rulesInRulestack.Items {
+			if ruleId, err := localrules.ParseLocalRuleID(pointer.From(v.Id)); err != nil {
+				if _, err := rulesClient.Delete(ctx, *ruleId); err != nil {
+					return fmt.Errorf("deleting rule %s from rulestack %s: %+v", ruleId, id, err)
+				}
+			}
+		}
+	}
+
+	// FQDN Lists
+	fqdnClient := client.ResourceManager.PaloAltoLocalRulestackFQDNClient
+	for _, rg := range rulestacks.Items {
+		rulestackId := fqdnlistlocalrulestack.NewLocalRulestackID(id.SubscriptionId, id.ResourceGroupName, pointer.From(rg.Name))
+		fqdnInRulestack, err := fqdnClient.ListByLocalRulestacksComplete(ctx, rulestackId)
+		if err != nil {
+			return fmt.Errorf("listing FQDNs for %s: %+v", id, err)
+		}
+		for _, v := range fqdnInRulestack.Items {
+			if fqdnId, err := localrules.ParseLocalRuleID(pointer.From(v.Id)); err != nil {
+				if _, err := rulesClient.Delete(ctx, *fqdnId); err != nil {
+					return fmt.Errorf("deleting fqdn %s from rulestack %s: %+v", fqdnId, id, err)
+				}
+			}
+		}
+	}
+
+	// Certificates
+	certClient := client.ResourceManager.PaloAltoLocalRulestackCertificatesClient
+	for _, rg := range rulestacks.Items {
+		// Remove inspection config - blocks removal of certs if referenced
+		rulestackId := certificateobjectlocalrulestack.NewLocalRulestackID(id.SubscriptionId, id.ResourceGroupName, pointer.From(rg.Name))
+		rs, err := rulestacksClient.Get(ctx, localrulestacks.LocalRulestackId(rulestackId))
+		if err != nil {
+			return err
+		}
+		sec := pointer.From(rs.Model.Properties.SecurityServices)
+		if pointer.From(sec.OutboundTrustCertificate) != "" || pointer.From(sec.OutboundUnTrustCertificate) != "" {
+			sec.OutboundTrustCertificate = nil
+			sec.OutboundUnTrustCertificate = nil
+			rs.Model.Properties.SecurityServices = pointer.To(sec)
+			if err = rulestacksClient.CreateOrUpdateThenPoll(ctx, localrulestacks.LocalRulestackId(rulestackId), *rs.Model); err != nil {
```

Comment:
there are no guarantees that'll work fwiw, we should new up a new ID instead:

```suggestion
			localRulestackId := localrulestacks.NewLocalRulestackID(rulestackId.SubscriptionId, rulestackId.ResourceGroupName, rulestackId.Name)
			if err = rulestacksClient.CreateOrUpdateThenPoll(ctx, localRulestackId, *rs.Model); err != nil {
```